awx_collection/plugins/modules/tower_job_template.py

```diff
@@ -415,19 +429,26 @@ def main():
             credentials = []
         credentials.append(credential)
 
+    new_fields = {}
+    search_fields = {'name': name}
+
+    # Attempt to look up the related items the user specified (these will fail the module if not found)
+    organization_id = None
+    organization = module.params.get('organization')
+    if organization:
+        organization_id = module.resolve_name_to_id('organizations', organization)
+        tower_version = module.get_endpoint('ping')['json']['version']
```

Comment:
I'm generally hesitant to issue an extra request for this purpose, although, we've recently made some changes so that we get it in the response header. Maybe we could develop a shared method which is a performant way to obtain/compare version. @john-westcott-iv I'm all ears if you have ideas for that.

My alternative would be to simply not make this request. Always include `organization` in the JT search payload if given, and document as a backward-incompatible change. This can still be avoided by users with older server versions by not specifying the parameter. Anyway, if they are facing that problem, they aren't matching server-collection versions, which we discourage.

Reply:
I'm all for not getting tower version, or just dumping this out and specifying it in docs. 
We could just rely on the user not to supply the org if they are prior to 3.7 and add a note in documentation. 
At the time I was trying to just make sure it would work against 3.7, 3.6 and awx. If that's the consensus, I'll remove the parts that the check version.

Reply:
We do pull the version out of the headers in module_util/tower_api.py around like 224. It should be fairly trivial to make properties in the object to hold these. On the other side, if we are going to have something that compares version we probably would want to make a method in the tower_api object because you would need to make it work for both tower an AWX.